pkg/scheduler/testing/framework_helpers.go

```diff
@@ -98,9 +98,6 @@ func RegisterPluginAsExtensionsWithWeight(pluginName string, weight int32, plugi
 			}
 			ps.Enabled = append(ps.Enabled, schedulerapi.Plugin{Name: pluginName, Weight: weight})
 		}
-		//lint:ignore SA4006 this value of pluginConfigs is never used.
```

Comment:
@supriya-premkumar we'll need to add this back. we should use `//nolint:ineffassign` to tell the linter to ignore this line

Reply:
( https://golangci-lint.run/usage/false-positives/ for info on how to suppress stuff )

Reply:
the existing code (before this PR) actually seems quite wrong... this is appending to a slice passed in as an argument, which does not effectively modify the arg passed in. @ahg-g @alculquicondor can you take a look at this? I think pluginConfigs has to be passed as `*[]schedulerapi.PluginConfig` for this to be effective

Reply:
That does seem wrong indeed, we should change `RegisterPluginFunc` to return `(plugins *schedulerapi.Plugins, pluginConfigs []schedulerapi.PluginConfig)` so that we keep the reference to the slices here: https://github.com/kubernetes/kubernetes/blame/master/pkg/scheduler/testing/framework_helpers.go#L31

But many unit tests can't possibly pass if this indeed didn't preserve properly append to the original passed slices, not sure what is happening...

Reply:
ok, so it seems that `PluginConfig` may not be used by any tests, tests are just dependent on default config and proper registration of the plugin itself in the `Plugins` slice (passed as reference to reference). We should fix this and add a test that relies on a modified config.

Reply:
Created #99318

Reply:
I guess for this PR that block of code can be removed since it has no impact as is, and #99318 will add it back in a proper way.

Reply:
Ack. I will not add `//nolint:ineffassign` for this one.

Reply:
I think this should be left as a failing check and #99318 prioritized

Reply:
can you revert this change to make sure it still shows up as a failing check?

Reply:
Done.